### PR TITLE
hotfix(coingecko): account for derived currencies with strict scope

### PR DESCRIPTION
**Pull request by james-a-morris**
Missed reference to matic in suggested-fees. This enabled a _select_ number of derived tokens.

### CODE REVIEW

File: api/_constants.ts [hunk 1]
```diff
@@ -159,6 +159,8 @@ export const BLOCK_TAG_LAG = -1;
 // Note: this is a small subset of all the supported base currencies, but since we don't expect to use the others,
 // we've decided to keep this list small for now.
 export const SUPPORTED_CG_BASE_CURRENCIES = new Set(["eth", "usd"]);
+// Note: this is a small set of currencies that the API will derive from the base currencies by using USD as an intermediary.
+export const SUPPORTED_CG_DERRIVED_CURRENCIES = new Set(["matic"]);
```
Comment (nicholaspai) [inline api/_constants.ts:163]: typo, 
```suggestion
export const SUPPORTED_CG_DERIVED_CURRENCIES = new Set(["matic"]);
```

File: api/coingecko.ts [hunk 2]
```diff
@@ -114,17 +123,32 @@ const handler = async (
...
+        TOKEN_SYMBOLS_MAP[
+          baseCurrency.toUpperCase() as keyof typeof TOKEN_SYMBOLS_MAP
+        ];
+      quotePrice = await getCachedTokenPrice(
+        token.addresses[CHAIN_IDs.MAINNET],
+        "usd"
+      );
+      quotePrecision = token.decimals;
+    }
+    price = Number((price / quotePrice).toFixed(quotePrecision));
```
Comment (pxrl) [inline api/coingecko.ts:150]: Is the final `Number()` conversion necessary? The division and subsequent use of `.toFixed()` should already imply that it's a number.

Comment (james-a-morris) [reply api/coingecko.ts:146]: The only thing about this is that the API on prod returns a number but removing the `Number` would would mean the return type is a string.

Comment (pxrl) [reply api/coingecko.ts:146]: ty - I missed that `Number.toFixed()` returns a string, but of course that makes sense.

**Review by pxrl — APPROVED** [review]
This tests OK for me :+1:

**Pull request merged**